### ISSUE TITLE
Missing Axis Lines in geom_col emf vs. png

### ISSUE THREAD

**Issue by awpsoras** [plfjohnson/devEMF#8]
Big fan of this package (and vector graphics!). Below is a script that replicates the issue. 

There is something about `geom_col` from ggplot2 that causes missing lines in exported emf images. In the script below you will see that both of the `geom_point` figures are drawn correctly, but the plot drawn with `geom_col`,  `col_emf_actual` (which uses actual data) has missing axis lines.

Interestingly, the sample data plot, `col_emf`, has no issues, but the test with actual data does. I could try to see where the cutoff is (if it comes down to n rows manifests the issue). 

```R

library(ggplot2)
library(here)
testset <- data.frame(
  mz = c(40, 80, 120),
  abundance = c(0.4, 0.5, 0.2)
)

actualset <- data.frame(
  mz = c(37, 47, 48, 49, 50, 51, 52, 61, 62, 63, 64, 65, 66, 70, 72, 73, 74, 75, 76, 77, 78, 82, 87, 88, 89, 90, 91, 92, 94, 96, 98, 99, 100, 101, 103, 105, 111, 112, 113, 119, 120, 123, 124, 125, 126, 127, 131, 135, 136, 137, 138, 139, 140, 141, 142, 154, 155, 156, 157, 158, 163, 167),
  abundance = c(0.1508364, 1195.918, 70.7237, 162.6667, 78.49703, 188.5357, 21.71536, 14.90221, 49.73117, 157.2265, 11.74636, 250.3775, 20.82964, 45.87701, 13.12837, 39.78162, 31.34705, 30.43889, 137.1578, 420.8023, 51.7579, 3.638256, 289.0885, 22.70313, 75.78088, 13.05543, 1370.906, 109.3451, 5.549341, 15.46811, 20.0336, 3.722722, 9.18993, 3.236995, 11.77533, 25.06347, 42.94441, 8.145709, 80.12592, 26.504, 0.913558, 54.98345, 60.42388, 94.35236, 8.102229, 10.84575, 6.904101, 45.25018, 6.247094, 74.20429, 39.85105, 9999, 1237.8, 377.3348, 27.04292, 2270.932, 292.2786, 108.8293, 12.86213, 23.54544, 7.166618, 3.249533)
)

# draws correctly
point_emf <- 
  testset |>
  ggplot(aes(x = mz, y = abundance)) +
  geom_point() +
  theme_bw() +
  theme(
    panel.grid = element_blank(),
    plot.background = element_rect(fill = NULL, color = "black")
  )

# draws correctly
col_emf <- 
  testset |>
  ggplot(aes(x = mz, y = abundance)) +
  geom_col(width = 0.25) +
  theme_bw() +
  theme(
    panel.grid = element_blank(),
    plot.background = element_rect(fill = NULL, color = "black")
  )

# draws correctly
point_emf_actual <- 
  actualset |>
  ggplot(aes(x = mz, y = abundance)) +
  geom_point() +
  theme_bw() +
  theme(
    panel.grid = element_blank(),
    plot.background = element_rect(fill = NULL, color = "black")
  )

# missing axis lines
col_emf_actual <- 
  actualset |>
  ggplot(aes(x = mz, y = abundance)) +
  geom_col(width = 0.25) +
  theme_bw() +
  theme(
    panel.grid = element_blank(),
    plot.background = element_rect(fill = NULL, color = "black")
  )

devEMF::emf(here("point_emf.emf"))
point_emf
dev.off()

devEMF::emf(here("col_emf.emf"))
col_emf
dev.off()

devEMF::emf(here("point_emf_actual.emf"))
point_emf_actual
dev.off()

devEMF::emf(here("col_emf_actual.emf"))
col_emf_actual
dev.off()

```

**Comment by awpsoras** [comment]
I did some further experimentation and it looks like removing a single row from either end of the dataframe allows the axis lines to be drawn in the resulting emf. So, slicing from `2:nrow(actualset)` or from `1:(nrow(actualset)-1)` both result in correct emf plots, which makes me think this is a scale issue.

I also tested sequential data up to 1000 rows with no issues, so this comes down to the actual dataset being plotted, not the number of rows (which again points towards axis scaling potentially being the culprit).

**Comment by plfjohnson** [comment]
Thank you for reporting this bug.  You've found an edge case that reveals a flaw in how I handle recycling entries in the EMF+ object table, which has a max of 64 entries used for paths, pens, and brushes; your hypothesis about the scale of the data being key is exactly right.  I'm going to have to think a little about how to best reengineer this recycling, but as a stopgap, you could ask devEMF to generate EMF instead of EMF+ output (oddly, EMF has a much higher limit to its object table)
```
devEMF::emf(..., emfPlus = FALSE)
```

**Comment by plfjohnson** [comment]
Should be fixed with commit 11047b99251e187cd71fabba2f48e46c70904f72 which changes the slot chosen for recycling in the EMF+ object table from created-longest-ago to used-longest-ago.  I plan to upload to cran as ```v4.5```.
The trigger for this bug was quite subtle, so much appreciate your clear report.

**Comment by awpsoras** [comment]
`emfPlus = FALSE` did work in the meantime, thank you! Also great to see it worked out - this package is a life saver for publications.
